docs/source/java/flight.rst

```diff
@@ -0,0 +1,302 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+================
+Arrow Flight RPC
+================
+
+Arrow Flight is an RPC framework for efficient transfer of Flight data
+over the network.
+
+.. contents::
+
+.. seealso::
+
+   :doc:`Flight protocol documentation <../format/Flight>`
+        Documentation of the Flight protocol, including how to use
+        Flight conceptually.
+
+   `Java Cookbook <https://arrow.apache.org/cookbook/java/flight.html>`_
+        Recipes for using Arrow Flight in Java.
+
+Writing a Flight Service
+========================
+
+Arrow Flight is a development framework to offer to us the building blocks
+to implement applications base on our needs.
+
+In this journey, you need to implement how do you are planning to expose
+your business operations in base of RPC Methods and Request Patterns
+provided by Flight.
+
+For Flight Server side operations you need to implement FlightProducer interface
+methods.
+
+.. code-block:: Java
+
+    public class TutorialFlightProducer implements FlightProducer {
+        @Override
+        // Override methods or use NoOpFlightProducer for only methods needed
+    }
+
```

Comment:
See #12815 where we explain ServerCallContext and errors here.

Reply:
Added